File: Working with Copy to Output Directory in C# and VB.NET/[C#][VB]-Working with Copy to Output Directory in C# and VB.NET/C# and VB/WindowsApplication1/DataGridViewMethods.vb
Module DataGridViewMethods
    <System.Diagnostics.DebuggerHidden()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Sub ExpandColumns(ByVal sender As DataGridView)
        For Each col As DataGridViewColumn In sender.Columns
            col.AutoSizeMode = DataGridViewAutoSizeColumnMode.AllCells
        Next
    End Sub
End Module

File: Working with Copy to Output Directory in C# and VB.NET/[C#][VB]-Working with Copy to Output Directory in C# and VB.NET/C# and VB/WindowsFormsApplication1/DataGridViewMethods.cs
using System.Windows.Forms;

namespace DataGridViewExtensionMethods
{
    /// <summary>
    /// Language extension methods for DataGridView
    /// </summary>
    
    public static class DataGridViewMethods
    {
        /// <summary>
        /// Set all columns AutoSizeMode to AllCells
        /// </summary>
        /// <param name="sender">DataGridView</param>
        /// <example>
        /// <code language='vbnet' title='VB.NET Example'>DataGridView1.ExpandColumns()</code>
        /// </example>
        [System.Diagnostics.DebuggerHidden]
        public static void ExpandColumns(this DataGridView sender)
        {
            foreach (DataGridViewColumn col in sender.Columns)
            {
                col.AutoSizeMode = DataGridViewAutoSizeColumnMode.AllCells;
            }
        }
    }
}